kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,187 @@
+//! Interface for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenPixelFormat {
+        /// No pixel format
+        None = 0,
```

Comment:
What would this be?

Reply:
I's a marker for an invalid pixel format. It is useful when a call to ``get_supported_pixel_format`` with an index out of bounds. It could change it and return an Option, maybe that would be a better way.